spec/cordova/plugin/save.spec.js

```diff
@@ -54,15 +61,145 @@ describe('cordova/plugin/save', function () {
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
     });
     describe('getSpec helper method', function () {
-        it('should return a plugin source\'s url or path property immediately');
-        it('should return a version if a version was provided to plugin id');
-        it('should return a version that includes scope if scope was part of plugin id');
-        it('should fall back to using PluginInfoProvider to retrieve a version as last resort');
+        it('should return a plugin source\'s url or path property immediately', function (done) {
+            var fake_fetch_json =
+                {'VRPlugin': {'source': {
+                    'url': 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git'
+                },
+                'is_top_level': true,
+                'variables': {
+                    'var 1': ' '
+                }}};
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            spyOn(save, 'getSpec');
+            save(projectRoot).then(function (result) {
+                expect(save.getSpec).toHaveBeenCalledWith(Object({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }), '/some/path', 'VRPlugin');
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
+
+        it('should return a version if a version was provided to plugin id', function (done) {
+            var fake_fetch_json =
+                {'cordova-plugin-camera': {'source': {
+                    'type': 'registry',
+                    'id': 'cordova-plugin-camera@^1.1.0'
+                },
+                'is_top_level': true }};
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            save(projectRoot).then(function () {
+                expect(save.versionString).toHaveBeenCalledWith('^1.1.0');
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
+
+        it('should return a version that includes scope if scope was part of plugin id', function (done) {
+            var fake_fetch_json =
+                {'cordova-plugin-camera': {'source': {
+                    'type': 'registry',
+                    'id': '@scoped/package@^1.0.0'
+                },
+                'is_top_level': true }};
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            spyOn(save, 'getSpec').and.callThrough();
+            save(projectRoot).then(function () {
+                expect(save.getSpec).toHaveBeenCalledWith(Object({ type: 'registry', id: '@scoped/package@^1.0.0' }), '/some/path', 'cordova-plugin-camera');
```

Comment:
Same, need to check what `save.getSpec` returns. First you need to change line 181 from `save(projectRoot)` to `save.getSpec({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }, '/some/path', 'VRPlugin')`